subprojects/build-cache/src/main/java/org/gradle/caching/internal/CacheableEntity.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.internal;
+
+/**
+ * An entity that can potentially be stored in the build cache.
+ */
+public interface CacheableEntity {
```

Comment:
We are using `toString` in some places to get a human readable name. Should this be a `Describable`? Or we should at least state here that types implementing the interface should have a nice `toString()` method.

Reply:
Yeah, I think we should make this `Describable`...